presto-main/src/main/java/com/facebook/presto/server/QueryEventLogger.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.event.query.QueryCompletionEvent;
+import com.facebook.presto.execution.QueryManagerConfig;
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.google.common.base.Throwables;
+import io.airlift.event.client.AbstractEventClient;
+import io.airlift.event.client.JsonEventSerializer;
+
+import javax.inject.Inject;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.logging.FileHandler;
+import java.util.logging.Formatter;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+public class QueryEventLogger
+        extends AbstractEventClient
+{
+    private final Logger logger = Logger.getLogger("QueryEventLogger");
+
+    private final JsonFactory factory = new JsonFactory();
+    private final JsonEventSerializer serializer = new JsonEventSerializer(QueryCompletionEvent.class);
+    private final String completedQueriesLogFile;
+
+    @Inject
+    public QueryEventLogger(QueryManagerConfig queryManagerConfig)
+    {
+        this.completedQueriesLogFile = queryManagerConfig.getCompletedQueriesLogFile();
+        if (completedQueriesLogFile != null) {
```

Comment:
Can you reverse it and return early?
E.g.
```java
if (completedQueriesLogFile == null) {
  return;
}
try ...
```

BTW shouldn't it be more explicit, e.g. by throwing an exception (to handle such a case somewhere where you expected the logger to be working) instead of silently doing nothing when there is no log file given?